grails-app/services/au/org/ala/GenerateService.groovy

```diff
@@ -110,7 +110,12 @@ class GenerateService {
                             //species image do not expire. when the image changes the url changes
                             cachedFile = new File(cacheDir + taxon.largeImageUrl.replaceAll("[^a-zA-Z0-9\\-\\_\\.]", "") + ".jpg")
                             if (!cachedFile.exists()) {
-                                FileUtils.copyURLToFile(new URL("${taxon.largeImageUrl.replace('raw', 'smallRaw')}"), cachedFile)
+                                try {
+                                    FileUtils.copyURLToFile(new URL("${taxon.largeImageUrl.replace('raw', 'smallRaw')}"), cachedFile)
+                                }
+                                catch (err){
+                                    log.error("failed to cache the image: " + taxon.largeImageUrl, err)
```

Comment:
- Use the actual URL in the message? `taxon.largeImageUrl.replace('raw', 'smallRaw')`
- Do not include the stack trace because it will only clutter the log.
- `err.message` may be useful as a replacement to the stack trace.

Reply:
Thanks Adam. I have fixed this.